action.yml

```diff
@@ -14,12 +14,29 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Resolve Golang version and set GOTOOLCHAIN
+      id: version
+      run: |
+        semver="${{ inputs.version }}"
+        if [[ -n "${{ inputs.version-file }}" ]]; then
+          goVersion=$(cat "${{ inputs.version-file }}" | grep --extended-regexp "^go +[0-9]")
+          # ignore PATCH component - only consider MAJOR.MINOR
+          if [[ $goVersion =~ ^go\ +([0-9]\.[0-9]+) ]]; then
```

Comment:
Use `grep` to extract out the `go X.XX.X` line from the asked `go.mod` - then reduce the version to just the `X.XX` component.